burn-tensor/src/tensor/backend/base.rs

```diff
@@ -101,21 +101,75 @@ pub(crate) type ADBackendTensorPrimitive<const D: usize, B> =
 
 /// Trait that allows a backend to support autodiff.
 pub trait ADBackend: Backend {
+    /// The inner backend type.
     type InnerBackend: Backend<Device = Self::Device, FloatElem = Self::FloatElem>;
+
+    /// Gradients type.
     type Gradients: Send + Sync;
 
+    /// Backward pass.
+    ///
+    /// # Arguments
+    ///
+    /// * `tensor` - The tensor to compute the gradients for.
+    ///
+    /// # Returns
+    ///
+    /// The gradients.
     fn backward<const D: usize>(tensor: Self::TensorPrimitive<D>) -> Self::Gradients;
+
+    /// Gradient of a tensor.
+    ///
+    /// # Arguments
+    ///
+    /// * `tensor` - The tensor to compute the gradient for.
+    ///
+    /// # Returns
+    ///
+    /// An optional tensor containing the gradient.
     fn grad<const D: usize>(
         tensor: &Self::TensorPrimitive<D>,
         grads: &Self::Gradients,
     ) -> Option<ADBackendTensorPrimitive<D, Self>>;
+
+    /// Removes the gradient of a tensor.
+    ///
+    /// # Arguments
+    ///
+    /// * `tensor` - The tensor to remove the gradient for.
+    /// * `grads` - The gradients.
```

Comment:
This is the same as the previous method, but removes the gradient from the pool (so that it can be used with inplace operations). So a `pop` instead of a `get`.